python/cudf/cudf/core/dataframe.py

```diff
@@ -5105,7 +5118,7 @@ def quantile(
         if isinstance(q, numbers.Number) and numeric_only:
             result = result.fillna(np.nan)
             result = result.iloc[0]
-            result.index = as_index(data_df.columns)
+            result.index = data_df._data.to_pandas_index()
```

Comment:
The original code seems to require a cudf index here?

Reply:
The index can be set with either a pandas or a cudf index object and both will work just fine. It ultimately just calls the `cudf.Index` constructor. Since the column data is on the host to start with, I don't think we'd benefit all that much if we tried to directly construct a cudf Index here and we may as well do it this way.